tiledb/sm/storage_manager/consolidator.cc

```diff
@@ -842,6 +859,12 @@ Status Consolidator::set_config(const Config* config) {
   RETURN_NOT_OK(merged_config.get<uint32_t>(
       "sm.consolidation.step_max_frags", &config_.max_frags_, &found));
   assert(found);
+  const char* mode;
```

Comment:
Let's use a `std::string` instead of a `char*` here:

```
const std::string mode = 
  merged_config.get("sm.consolidation.mode", &found);
if (!found)
  ...
config_.mode_ = mode;
```

Reply:
I think the way it is right now is easier and faster, as we simply return the constant char pointer of the string stored internally for the mode. If we return `std::string` we will need to instantiate a new template function for that, as it is currently not supported.

Reply:
I must be missing something because I'm able to load directly into a string without a template type here:
https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/filesystem/azure.cc#L74

Reply:
I see the `Config::get` function now, I will switch to that.